core/src/test/java/org/apache/accumulo/core/data/RowRangeTest.java

```diff
@@ -0,0 +1,569 @@
+    void testAllEquality() {
+      RowRange range1 = RowRange.all();
+      RowRange range2 = RowRange.range((Text) null, false, null, false);
+
+      assertTrue(range1.equals(range2));
+    }
+  }
+
+  @Nested
+  class EqualsTests {
+
+    @Test
+    void testEqualsWithDifferentRanges() {
+      RowRange range1 = RowRange.closedOpen("r1", "row5");
+      RowRange range2 = RowRange.closedOpen("r2", "row4");
+      assertFalse(range1.equals(range2));
+    }
+
+    @Test
+    void testEqualsWithSameRange() {
+      RowRange range1 = RowRange.closedOpen("r1", "row5");
+      RowRange range2 = RowRange.closedOpen("r1", "row5");
+      assertTrue(range1.equals(range2));
+    }
+
+    @Test
+    void testEqualsWithDifferentStartRowInclusiveness() {
+      RowRange range1 = RowRange.closedOpen("r1", "row5");
+      RowRange range2 = RowRange.openClosed("r1", "row5");
+      assertFalse(range1.equals(range2));
+    }
+
+    @Test
+    void testEqualsWithDifferentEndRowInclusiveness() {
+      RowRange range1 = RowRange.closedOpen("r1", "row5");
+      RowRange range2 = RowRange.closedOpen("r1", "row4");
+      assertFalse(range1.equals(range2));
+    }
+
+    @Test
+    void testEqualsWithDifferentStartRowAndEndRowInclusiveness() {
+      RowRange range1 = RowRange.closedOpen("r1", "row5");
+      RowRange range2 = RowRange.openClosed("r1", "row4");
+      assertFalse(range1.equals(range2));
+    }
+
+    @Test
+    void testOverloadEquality() {
+      RowRange range1 = RowRange.closedOpen("r1", "row5");
+      RowRange range2 = RowRange.closedOpen(new Text("r1"), new Text("row5"));
+      RowRange range3 = RowRange.atLeast("row8");
+      RowRange range4 = RowRange.atLeast(new Text("row8"));
+      RowRange range5 = RowRange.lessThan("r2");
+      RowRange range6 = RowRange.lessThan(new Text("r2"));
+      RowRange range7 = RowRange.atMost("r3");
+      RowRange range8 = RowRange.atMost(new Text("r3"));
+
+      // Test that all ranges created using different entry point methods are equal
+      assertTrue(range1.equals(range2));
+      assertTrue(range3.equals(range4));
+      assertTrue(range5.equals(range6));
+      assertTrue(range7.equals(range8));
+
+      // Test that ranges with different properties are not equal
+      assertFalse(range1.equals(range3));
+      assertFalse(range1.equals(range5));
+      assertFalse(range1.equals(range7));
+      assertFalse(range3.equals(range5));
+      assertFalse(range3.equals(range7));
+      assertFalse(range5.equals(range7));
+    }
+  }
+
+  @Nested
+  class CompareToTests {
+
+    @Test
+    void testCompareWithSameRange() {
+      RowRange range1 = RowRange.open("r1", "r3");
+      RowRange range2 = RowRange.open("r1", "r3");
+      assertEquals(0, range1.compareTo(range2));
+    }
+
+    @Test
+    void testCompareWithDifferentStartRow() {
+      RowRange range1 = RowRange.open("r1", "r3");
+      RowRange range2 = RowRange.open("r2", "r3");
+      assertTrue(range1.compareTo(range2) < 0);
```

Comment:
For all of the compare to test it would be nice to try both directions.

```suggestion
      assertTrue(range1.compareTo(range2) < 0);
      assertTrue(range2.compareTo(range1) > 0);
```